### ISSUE TITLE
Loading ttbin, exporting GPX

### ISSUE THREAD

**Issue by Jacquers** [scubajorgen/TomTomWatch#6]
Thanks for this utility, I was able to get the ttbin files off my older Multisport watch :)

Loading a ttbin file into the program doesn't seem to work (maybe because my watch wasn't connected?), although it works in the older converter program, but the gpx file it saves doesn't contain any coordinate data.

Attached are the files that TomTom MySports saved if you need an example. I'm guessing the older watch doesn't have all the fields that the newer one has.

Any help would be appreciated, thanks.

[2018-02-21.zip](https://github.com/scubajorgen/TomTomWatch/files/1751707/2018-02-21.zip)

**Comment by scubajorgen** [comment]
Thanx for the feedback.
Yes, the watch needs to be connected when loading a ttbin file. That does not make sense.  I have to change this.

Apart from this issue the ttbin loading and gpx export goes fine (when a watch is connected). Indeed, your ttbin is an older ttbin version (0x07, whereas latest is 0x0a). But this older version is supported since version 1.2 of TomTomWatch. So if you encounter trouble loading the ttbin and exporting the gpx (when a watch is connected) please check the version of TomTomWatch you are using.
![run](https://user-images.githubusercontent.com/26226586/36604125-1efe7742-18bd-11e8-9870-bbecdd7bfe76.png)

**Comment by scubajorgen** [comment]
Solved.

**Comment by Jacquers** [comment]
Thanks very much :)

**Comment by Jacquers** [comment]
I downloaded 2 activities from my watch using the official TT software, which I have set to save the ttbin, fit and gpx files. I closed the software and opened TomTomWatch 1.3 which was now able to load the ttbin files with the watch connected. It saves the GPX but it's only a 1 kb file. Any help would be appreciated.

**Comment by scubajorgen** [comment]
Thanx for the feedback.

Actually another bug. If no height values are present, which is generally the case for all TomTom watches except the Adventurer, empty GPX files are written. I only tested it with Google height service enabled, then the problem does not occur (with this feature height values are added by the TomTomWatch software using the Google elevation service). I solved the issue. For your ease of use I included the jar file of 1.4 in which the issues are solved. Just rename .zip to .jar and replace the 1.3 jar with this one.

[TomTomWatch-1.4.zip](https://github.com/scubajorgen/TomTomWatch/files/1772713/TomTomWatch-1.4.zip)
